runtime/gc_vlhgc/AllocationContextBalanced.cpp

```diff
@@ -1103,6 +1103,59 @@ MM_AllocationContextBalanced::setNumaAffinityForThread(MM_EnvironmentBase *env)
 }
 
 #if defined(J9VM_GC_SPARSE_HEAP_ALLOCATION)
+bool
+MM_AllocationContextBalanced::needAllocateReservedRegionFraction(MM_EnvironmentBase *env, uintptr_t fraction)
+{
+	const uintptr_t regionSize = _heapRegionManager->getRegionSize();
+	Assert_MM_true((0 < fraction) && (regionSize > fraction));
+
+	bool ret = false;
+	fraction = MM_Math::roundToCeiling(J9_GC_MINIMUM_OBJECT_SIZE, fraction);
+
+	if (regionSize == fraction) {
+		ret = true;
+	} else {
+		lockCommon();
+		if (0 == _fractionReservedRegion) {
+			_fractionReservedRegion = fraction;
+			ret = true;
+		} else {
+			_fractionReservedRegion += fraction;
+			if ((intptr_t)regionSize <= _fractionReservedRegion) {
+				_fractionReservedRegion -= regionSize;
+				ret = true;
+			}
+		}
+		unlockCommon();
+	}
+	return ret;
+}
+
+bool
+MM_AllocationContextBalanced::needRecycleReservedRegionFraction(MM_EnvironmentBase *env, uintptr_t fraction)
+{
+	const uintptr_t regionSize = _heapRegionManager->getRegionSize();
+	Assert_MM_true((0 < fraction) && (regionSize > fraction));
+
+	bool ret = false;
+	fraction = MM_Math::roundToCeiling(J9_GC_MINIMUM_OBJECT_SIZE, fraction);
+
+	if (regionSize == fraction) {
+		ret = true;
+	} else {
+		lockCommon();
+		_fractionReservedRegion -= fraction;
+		if (0 >= _fractionReservedRegion) {
```

Comment:
I prefer _fractionReservedRegion being unsigned, but doing the size check (against regionSize) before doing the subtraction.